lib/phoenix/live_dashboard/pages/os_mon_page.ex

```diff
@@ -26,105 +26,53 @@ defmodule Phoenix.LiveDashboard.OSMonPage do
 
   @impl true
   def render_page(assigns) do
+    # FIXME this is wrong we should do it in mount
     os_mon = SystemInfo.fetch_os_mon_info(assigns.page.node)
     cpu_count = length(os_mon.cpu_per_core)
-    row_params = %{os_mon: os_mon, cpu_count: cpu_count, csp_nonces: assigns.csp_nonces}
+    assigns = assign(assigns, os_mon: os_mon, cpu_count: cpu_count)
 
-    top_row = cpu_components(row_params) ++ memory_components(row_params)
-    bottom_row = [disk_usage_row(row_params)]
-
-    row(
-      components: [
-        columns(components: top_row),
-        columns(components: bottom_row)
-      ]
-    )
+    ~H"""
+    <.row>
+      <:col>
+        <.cpu_components {assigns}/>
+      </:col>
+      <:col>
+        <.memory_components {assigns}/>
+      </:col>
+    </.row>
+    <.row>
+      <:col>
+        <.disk_usage_row {assigns}/>
+      </:col>
+    </.row>
+    """
   end
 
-  defp cpu_components(%{os_mon: %{cpu_avg1: num1, cpu_avg5: num5, cpu_avg15: num15}} = row_params)
+  defp cpu_components(%{os_mon: %{cpu_avg1: num1, cpu_avg5: num5, cpu_avg15: num15}} = assigns)
        when is_number(num1) and is_number(num5) and is_number(num15) do
-    [
-      [cpu_load_row(row_params)] ++
-        if(row_params.cpu_count > 0, do: [cpu_avg_row(row_params)], else: [])
-    ]
+    ~H"""
+    <.cpu_load_row {assigns}/>
```

Comment:
Passing assigns to components is almost always a smell. For example, here we just use `os_mon`. Can we instead pass it explicitly around?

```suggestion
    <.cpu_load_row os_mon={@os_mon} />
```

The same for the other components in this file.